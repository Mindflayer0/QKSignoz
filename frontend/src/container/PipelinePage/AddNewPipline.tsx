import { PlusOutlined } from '@ant-design/icons';
import TextToolTip from 'components/TextToolTip';
import React from 'react';
import { useTranslation } from 'react-i18next';

import { Button, ButtonContainer } from './styles';

function AddNewPipline({ setActionType }: AddNewPiplineType): JSX.Element {
	const { t } = useTranslation(['pipeline']);

	return (
		<ButtonContainer>
			<TextToolTip text="Add Piplines" />
			<Button
				icon={<PlusOutlined />}
				onClick={(): void => setActionType('add-pipeline')}
				type="primary"
			>
				{t('new_pipeline')}
			</Button>
		</ButtonContainer>
	);
}

interface AddNewPiplineType {
	setActionType: (b?: string) => void;
}

export default AddNewPipline;
