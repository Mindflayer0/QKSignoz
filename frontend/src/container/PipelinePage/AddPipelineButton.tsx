import { PlusOutlined } from '@ant-design/icons';
import TextToolTip from 'components/TextToolTip';
import React from 'react';
import { useTranslation } from 'react-i18next';

import { Button, ButtonContainer } from './styles';

function AddPipelineButton({
	setActionType,
}: AddPipelineButtonType): JSX.Element {
	const { t } = useTranslation(['pipeline']);

	return (
		<ButtonContainer>
			<TextToolTip text={t('add_new_pipeline')} />
			<Button
				icon={<PlusOutlined />}
				onClick={(): void => setActionType('add-pipeline')}
				type="primary"
			>
				{t('new_pipeline')}
			</Button>
		</ButtonContainer>
	);
}

interface AddPipelineButtonType {
	setActionType: (b?: string) => void;
}

export default AddPipelineButton;
