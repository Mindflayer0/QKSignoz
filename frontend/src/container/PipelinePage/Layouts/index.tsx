import React, { useState } from 'react';

import PipelineListsView from '../PipelineListsView';
import CreatePipelineButton from './CreatePipelineButton';
import PiplinesSearchSection from './PiplinesSearchSection';

function PipelinePageLayout(): JSX.Element {
	const [isActionType, setActionType] = useState<string>();
	return (
		<>
			<CreatePipelineButton setActionType={setActionType} />
			<PiplinesSearchSection />
			<PipelineListsView
				isActionType={isActionType as string}
				setActionType={setActionType}
			/>
		</>
	);
}

export enum ActionType {
	AddPipeline = 'add-pipeline',
	EditPipeline = 'edit-pipeline',
	AddProcessor = 'add-processor',
	EditProcessor = 'edit-processor',
}
export default PipelinePageLayout;
