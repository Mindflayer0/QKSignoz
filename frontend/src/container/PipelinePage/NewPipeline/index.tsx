/* eslint-disable no-template-curly-in-string */
import { Button, Divider, Form, Input, Modal, Typography } from 'antd';
import React from 'react';

import PiplinesSearchBar from '../SearchBar';
import { inputfiledName } from './utils';

function NewPipline({
	addPipeline,
	setNewAddPiplines,
}: {
	addPipeline: boolean;
	setNewAddPiplines: (b: boolean) => void;
}): JSX.Element {
	const [form] = Form.useForm();

	const validateMessages = {
		required: '${label} is required!',
		types: {
			email: '${label} is not a valid email!',
			number: '${label} is not a valid number!',
		},
		number: {
			range: '${label} must be between ${min} and ${max}',
		},
	};

	const onFinish = (): void => {
		console.log('finish');
	};

	return (
		<Modal
			title={<Typography.Title level={4}>Create Pipeline</Typography.Title>}
			centered
			open={addPipeline}
			onOk={(): void => setNewAddPiplines(false)}
			onCancel={(): void => setNewAddPiplines(false)}
			width={800}
			footer={[
				<Button key="back" onClick={(): void => setNewAddPiplines(false)}>
					Cancel
				</Button>,
				<Button
					key="submit"
					type="primary"
					onClick={(): void => setNewAddPiplines(false)}
				>
					Create
				</Button>,
			]}
		>
			<Divider plain />
			<div style={{ marginTop: '25px' }}>
				<span>Filter</span>
				<div style={{ marginTop: '5px' }}>
					<PiplinesSearchBar />
				</div>
			</div>
			<Form
				form={form}
				layout="vertical"
				style={{ marginTop: '20px' }}
				onFinish={onFinish}
				validateMessages={validateMessages}
			>
				{inputfiledName.map((i) => {
					if (i.id === 3) {
						return (
							<Form.Item
								name={['user', 'name']}
								label={i.fildName}
								key={i.id}
								rules={[{ required: true }]}
							>
								<Input.TextArea rows={2} placeholder={i.placeholder} maxLength={4} />
							</Form.Item>
						);
					}
					return (
						<Form.Item
							name={['user', 'description']}
							label={i.fildName}
							key={i.id}
							rules={[{ required: true }]}
						>
							<Input placeholder={i.placeholder} />
						</Form.Item>
					);
				})}
			</Form>
			<Divider plain />
		</Modal>
	);
}

export default NewPipline;
