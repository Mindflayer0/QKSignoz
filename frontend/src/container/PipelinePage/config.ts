import { themeColors } from 'constants/theme';

const positionCss: React.CSSProperties['position'] = 'relative';

export const iconStyle = { color: themeColors.gainsboro, fontSize: '1.12rem' };

export const ListiconStyle = {
	backgroundColor: themeColors.navyBlue,
	fontSize: '14px',
};

export const modalIconStyle = {
	backgroundColor: themeColors.navyBlue,
};

export const modalIcon = {
	backgroundColor: themeColors.navyBlue,
	position: positionCss,
	top: '10px',
};

export const deleteModalDescriptionStyle = {
	fontWeight: 400,
	fontStyle: 'normal',
	fontSize: '12px',
};

export const modalFooterStyle = {
	display: 'flex',
	gap: '8px',
	marginLeft: '15px',
	alignItems: 'center',
};

export const modalTitleStyle = {
	fontStyle: 'normal',
	fontWeight: 600,
	fontSize: '14px',
	lineHeight: '22px',
};

export const sublistDataStyle = {
	backgroundColor: themeColors.navyBlue,
	height: '22px',
	width: '22px',
};

export const modalFooterTitle = {
	fontStyle: 'normal',
	fontWeight: 400,
	fontSize: '12px',
	lineHeight: '20px',
};
