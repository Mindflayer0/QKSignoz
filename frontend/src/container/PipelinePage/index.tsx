import React, { useState } from 'react';

import AddPipelineButton from './AddPipelineButton';
import ListOfPipelines from './ListOfPipelines';
import PiplinesSearchBar from './SearchBar';

function PipelinePage(): JSX.Element {
	const [isActionType, setActionType] = useState<string>();
	return (
		<>
			<AddPipelineButton setActionType={setActionType} />
			<PiplinesSearchBar />
			<ListOfPipelines
				isActionType={isActionType as string}
				setActionType={setActionType}
			/>
		</>
	);
}

export enum ActionType {
	AddPipeline = 'add-pipeline',
	EditPipeline = 'edit-pipeline',
	AddProcessor = 'add-processor',
	EditProcessor = 'edit-processor',
}
export default PipelinePage;
