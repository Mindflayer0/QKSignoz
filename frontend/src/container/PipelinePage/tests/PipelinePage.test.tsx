import { render } from '@testing-library/react';
import PipelinePage from 'container/PipelinePage';
import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import store from 'store';

describe('PipelinePage test', () => {
	it('should render PipelinePage', () => {
		const { asFragment } = render(
			<MemoryRouter>
				<Provider store={store}>
					<PipelinePage />
				</Provider>
			</MemoryRouter>,
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
