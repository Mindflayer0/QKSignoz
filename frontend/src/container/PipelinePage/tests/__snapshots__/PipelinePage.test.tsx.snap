// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PipelinePage test should render PipelinePage 1`] = `
<DocumentFragment>
  .c0.c0.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  margin-bottom: 2rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1.c1.c1 {
  margin-left: 1rem;
}

<div
    class="c0"
  >
    <span
      aria-label="question-circle"
      class="anticon anticon-question-circle"
      role="img"
      style="font-size: 1.3125rem;"
    >
      <svg
        aria-hidden="true"
        data-icon="question-circle"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 708c-22.1 0-40-17.9-40-40s17.9-40 40-40 40 17.9 40 40-17.9 40-40 40zm62.9-219.5a48.3 48.3 0 00-30.9 44.8V620c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8v-21.5c0-23.1 6.7-45.9 19.9-64.9 12.9-18.6 30.9-32.8 52.1-40.9 34-13.1 56-41.6 56-72.7 0-44.1-43.1-80-96-80s-96 35.9-96 80v7.6c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V420c0-39.3 17.2-76 48.4-103.3C430.4 290.4 470 276 512 276s81.6 14.5 111.6 40.7C654.8 344 672 380.7 672 420c0 57.8-38.1 109.8-97.1 132.5z"
        />
      </svg>
    </span>
    <button
      class="ant-btn css-dev-only-do-not-override-1i536d8 ant-btn-primary c1"
      type="button"
    >
      <span
        aria-label="plus"
        class="anticon anticon-plus"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="plus"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <defs>
            <style />
          </defs>
          <path
            d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
          />
          <path
            d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
          />
        </svg>
      </span>
      <span>
        new_pipeline
      </span>
    </button>
  </div>
  <span
    class="ant-input-group-wrapper ant-input-search css-dev-only-do-not-override-1i536d8"
  >
    <span
      class="ant-input-wrapper ant-input-group css-dev-only-do-not-override-1i536d8"
    >
      <span
        class="ant-input-affix-wrapper css-dev-only-do-not-override-1i536d8"
      >
        <input
          class="ant-input css-dev-only-do-not-override-1i536d8"
          placeholder="search_pipeline_placeholder"
          type="text"
          value=""
        />
        <span
          class="ant-input-suffix"
        >
          <span
            class="ant-input-clear-icon ant-input-clear-icon-hidden"
            role="button"
            tabindex="-1"
          >
            <span
              aria-label="close-circle"
              class="anticon anticon-close-circle"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="close-circle"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                />
              </svg>
            </span>
          </span>
        </span>
      </span>
      <span
        class="ant-input-group-addon"
      >
        <button
          class="ant-btn css-dev-only-do-not-override-1i536d8 ant-btn-default ant-btn-icon-only ant-input-search-button"
          type="button"
        >
          <span
            aria-label="search"
            class="anticon anticon-search"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="search"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
              />
            </svg>
          </span>
        </button>
      </span>
    </span>
  </span>
  .c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.5rem;
  margin-left: 5.875rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.25rem;
}

.c1 {
  color: #DBDBDB;
  font-size: 1.5rem;
}

.c2 {
  color: #DBDBDB;
  font-size: 1.5rem;
}

.c3 {
  color: #DBDBDB;
  font-size: 1.5rem;
}

.c5 {
  color: #8DCFF8;
  font-size: 1.12rem;
}

.c0 {
  margin-top: 3rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 1.25rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
    <div
      class="c0"
    >
      <div
        class="ant-table-wrapper css-dev-only-do-not-override-1i536d8"
      >
        <div
          class="ant-spin-nested-loading css-dev-only-do-not-override-1i536d8"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-table"
            >
              <div
                class="ant-table-container"
              >
                <div
                  class="ant-table-content"
                >
                  <table
                    style="table-layout: auto;"
                  >
                    <colgroup>
                      <col
                        class="ant-table-expand-icon-col"
                      />
                    </colgroup>
                    <thead
                      class="ant-table-thead"
                    >
                      <tr>
                        <th
                          class="ant-table-cell ant-table-row-expand-icon-cell"
                        />
                        <th
                          class="ant-table-cell"
                        />
                        <th
                          class="ant-table-cell"
                        >
                          Pipeline Name
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          Filters
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          Tags
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          Last Edited
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          Edited By
                        </th>
                        <th
                          class="ant-table-cell"
                          style="text-align: center;"
                        >
                          Actions
                        </th>
                        <th
                          class="ant-table-cell"
                        />
                      </tr>
                    </thead>
                    <tbody
                      class="ant-table-tbody"
                    >
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="1"
                        draggable="true"
                      >
                        <td
                          class="ant-table-cell ant-table-row-expand-icon-cell"
                        >
                          <span
                            aria-label="right"
                            class="anticon anticon-right"
                            role="img"
                            tabindex="-1"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="right"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                              />
                            </svg>
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <span
                            class="ant-avatar ant-avatar-sm ant-avatar-circle css-dev-only-do-not-override-1i536d8"
                            style="background: rgb(22, 104, 220);"
                          >
                            <span
                              class="ant-avatar-string"
                              style="transform: scale(1) translateX(-50%);"
                            >
                              1
                            </span>
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <span>
                            Apache common parser
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <span>
                            attributes.source == nginx
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <span>
                            <span
                              class="ant-tag ant-tag-magenta css-dev-only-do-not-override-1i536d8"
                            >
                              server
                            </span>
                            <span
                              class="ant-tag ant-tag-magenta css-dev-only-do-not-override-1i536d8"
                            >
                              app
                            </span>
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <span>
                            Mon Feb 06 2023
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <span>
                            nitya
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                          style="text-align: center;"
                        >
                          <div
                            class="ant-space css-dev-only-do-not-override-1i536d8 ant-space-horizontal ant-space-align-center"
                          >
                            <div
                              class="ant-space-item"
                              style="margin-right: 16px;"
                            >
                              <span>
                                <span
                                  aria-label="edit"
                                  class="anticon anticon-edit c1"
                                  role="img"
                                  tabindex="-1"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="edit"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </div>
                            <div
                              class="ant-space-item"
                              style="margin-right: 16px;"
                            >
                              <span>
                                <span
                                  aria-label="eye"
                                  class="anticon anticon-eye c2"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="eye"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M396 512a112 112 0 10224 0 112 112 0 10-224 0zm546.2-25.8C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM508 688c-97.2 0-176-78.8-176-176s78.8-176 176-176 176 78.8 176 176-78.8 176-176 176z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </div>
                            <div
                              class="ant-space-item"
                            >
                              <span>
                                <span
                                  aria-label="delete"
                                  class="anticon anticon-delete c3"
                                  role="img"
                                  tabindex="-1"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="delete"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M864 256H736v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zm-200 0H360v-72h304v72z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </div>
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="c4"
                          >
                            <span>
                              <button
                                aria-checked="false"
                                class="ant-switch css-dev-only-do-not-override-1i536d8"
                                role="switch"
                                type="button"
                              >
                                <div
                                  class="ant-switch-handle"
                                />
                                <span
                                  class="ant-switch-inner"
                                >
                                  <span
                                    class="ant-switch-inner-checked"
                                  />
                                  <span
                                    class="ant-switch-inner-unchecked"
                                  />
                                </span>
                              </button>
                            </span>
                            <span
                              style="cursor: move;"
                            >
                              <span
                                aria-label="holder"
                                class="anticon anticon-holder c5"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="holder"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M300 276.5a56 56 0 1056-97 56 56 0 00-56 97zm0 284a56 56 0 1056-97 56 56 0 00-56 97zM640 228a56 56 0 10112 0 56 56 0 00-112 0zm0 284a56 56 0 10112 0 56 56 0 00-112 0zM300 844.5a56 56 0 1056-97 56 56 0 00-56 97zM640 796a56 56 0 10112 0 56 56 0 00-112 0z"
                                  />
                                </svg>
                              </span>
                            </span>
                          </div>
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="2"
                        draggable="true"
                      >
                        <td
                          class="ant-table-cell ant-table-row-expand-icon-cell"
                        >
                          <span
                            aria-label="right"
                            class="anticon anticon-right"
                            role="img"
                            tabindex="-1"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="right"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                              />
                            </svg>
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <span
                            class="ant-avatar ant-avatar-sm ant-avatar-circle css-dev-only-do-not-override-1i536d8"
                            style="background: rgb(22, 104, 220);"
                          >
                            <span
                              class="ant-avatar-string"
                              style="transform: scale(1) translateX(-50%);"
                            >
                              2
                            </span>
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <span>
                            myservice trace parser
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <span>
                            attributes.source == myservice
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <span>
                            <span
                              class="ant-tag ant-tag-magenta css-dev-only-do-not-override-1i536d8"
                            >
                              server
                            </span>
                            <span
                              class="ant-tag ant-tag-magenta css-dev-only-do-not-override-1i536d8"
                            >
                              app
                            </span>
                            <span
                              class="ant-tag ant-tag-magenta css-dev-only-do-not-override-1i536d8"
                            >
                              host
                            </span>
                            <span
                              class="ant-tag ant-tag-magenta css-dev-only-do-not-override-1i536d8"
                            >
                              realse
                            </span>
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <span>
                            Sun Feb 05 2023
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <span>
                            nityananda
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                          style="text-align: center;"
                        >
                          <div
                            class="ant-space css-dev-only-do-not-override-1i536d8 ant-space-horizontal ant-space-align-center"
                          >
                            <div
                              class="ant-space-item"
                              style="margin-right: 16px;"
                            >
                              <span>
                                <span
                                  aria-label="edit"
                                  class="anticon anticon-edit c1"
                                  role="img"
                                  tabindex="-1"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="edit"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </div>
                            <div
                              class="ant-space-item"
                              style="margin-right: 16px;"
                            >
                              <span>
                                <span
                                  aria-label="eye"
                                  class="anticon anticon-eye c2"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="eye"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M396 512a112 112 0 10224 0 112 112 0 10-224 0zm546.2-25.8C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM508 688c-97.2 0-176-78.8-176-176s78.8-176 176-176 176 78.8 176 176-78.8 176-176 176z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </div>
                            <div
                              class="ant-space-item"
                            >
                              <span>
                                <span
                                  aria-label="delete"
                                  class="anticon anticon-delete c3"
                                  role="img"
                                  tabindex="-1"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="delete"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M864 256H736v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zm-200 0H360v-72h304v72z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </div>
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="c4"
                          >
                            <span>
                              <button
                                aria-checked="false"
                                class="ant-switch css-dev-only-do-not-override-1i536d8"
                                role="switch"
                                type="button"
                              >
                                <div
                                  class="ant-switch-handle"
                                />
                                <span
                                  class="ant-switch-inner"
                                >
                                  <span
                                    class="ant-switch-inner-checked"
                                  />
                                  <span
                                    class="ant-switch-inner-unchecked"
                                  />
                                </span>
                              </button>
                            </span>
                            <span
                              style="cursor: move;"
                            >
                              <span
                                aria-label="holder"
                                class="anticon anticon-holder c5"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="holder"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M300 276.5a56 56 0 1056-97 56 56 0 00-56 97zm0 284a56 56 0 1056-97 56 56 0 00-56 97zM640 228a56 56 0 10112 0 56 56 0 00-112 0zm0 284a56 56 0 10112 0 56 56 0 00-112 0zM300 844.5a56 56 0 1056-97 56 56 0 00-56 97zM640 796a56 56 0 10112 0 56 56 0 00-112 0z"
                                  />
                                </svg>
                              </span>
                            </span>
                          </div>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
              <div
                class="ant-table-footer"
              >
                <button
                  class="ant-btn css-dev-only-do-not-override-1i536d8 ant-btn-link c6"
                  type="button"
                >
                  <span
                    aria-label="plus"
                    class="anticon anticon-plus"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="plus"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <defs>
                        <style />
                      </defs>
                      <path
                        d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
                      />
                      <path
                        d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
                      />
                    </svg>
                  </span>
                  <span>
                    add_new_pipeline
                  </span>
                </button>
              </div>
            </div>
            <ul
              class="ant-pagination ant-table-pagination ant-table-pagination-right css-dev-only-do-not-override-1i536d8"
            >
              <li
                aria-disabled="true"
                class="ant-pagination-prev ant-pagination-disabled"
                title="Previous Page"
              >
                <button
                  class="ant-pagination-item-link"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <span
                    aria-label="left"
                    class="anticon anticon-left"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="left"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                      />
                    </svg>
                  </span>
                </button>
              </li>
              <li
                class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
                tabindex="0"
                title="1"
              >
                <a
                  rel="nofollow"
                >
                  1
                </a>
              </li>
              <li
                aria-disabled="true"
                class="ant-pagination-next ant-pagination-disabled"
                title="Next Page"
              >
                <button
                  class="ant-pagination-item-link"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <span
                    aria-label="right"
                    class="anticon anticon-right"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="right"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                      />
                    </svg>
                  </span>
                </button>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
